tests/runfiles/runfiles_test.py

```diff
@@ -525,6 +525,259 @@ def testDirectoryBasedRlocationWithRepoMappingFromOtherRepo(self) -> None:
+        repo_mapping = self._createTestRepositoryMapping()
+        
+        # Test exact lookups
+        self.assertTrue(("", "my_workspace") in repo_mapping)
+        self.assertTrue(("deps+specific_repo", "external_dep") in repo_mapping)
+        
+        # Test prefix-based lookups (consistent with __getitem__)
+        self.assertTrue(("deps+some_repo", "external_dep") in repo_mapping)
+        
+        # Test non-existent mapping
+        self.assertFalse(("nonexistent", "repo") in repo_mapping)
+
+    def testRepositoryMappingGet(self) -> None:
+        """Test get() method - should work for both exact and prefix-based mappings."""
+        repo_mapping = self._createTestRepositoryMapping()
+        
+        # Test exact lookups
+        self.assertEqual(repo_mapping.get(("", "my_workspace")), "_main")
+        self.assertEqual(repo_mapping.get(("deps+specific_repo", "external_dep")), "external_dep~exact")
+        
+        # Test prefix-based lookups
+        self.assertEqual(repo_mapping.get(("deps+some_repo", "external_dep")), "external_dep~prefix")
+        
+        # Test non-existent mapping
+        self.assertIsNone(repo_mapping.get(("nonexistent", "repo")))
+        self.assertEqual(repo_mapping.get(("nonexistent", "repo"), "default"), "default")
+
+    def testRepositoryMappingKeysValuesItems(self) -> None:
+        """Test keys(), values(), and items() methods (inherited from Mapping)."""
+        repo_mapping = self._createTestRepositoryMapping()
+        
+        # Test items()
+        items = list(repo_mapping.items())
+        self.assertEqual(len(items), 5)  # 3 exact + 2 prefixed
+        self.assertIn((("", "my_workspace"), "_main"), items)
+        self.assertIn((("deps+", "external_dep"), "external_dep~prefix"), items)
+        
+        # Test keys()
+        keys = list(repo_mapping.keys())
+        self.assertEqual(len(keys), 5)  # 3 exact + 2 prefixed
+        self.assertIn(("", "my_workspace"), keys)
+        self.assertIn(("deps+", "external_dep"), keys)
+        
+        # Test values()
+        values = list(repo_mapping.values())
+        self.assertEqual(len(values), 5)  # 3 exact + 2 prefixed
+        self.assertIn("_main", values)
+        self.assertIn("external_dep~prefix", values)
+
+    def testRepositoryMappingTruthiness(self) -> None:
+        """Test pythonic truthiness - non-empty mapping is truthy, empty mapping is falsy."""
+        repo_mapping = self._createTestRepositoryMapping()
+        
+        # Test non-empty mapping is truthy
+        self.assertTrue(repo_mapping)
+        self.assertTrue(bool(repo_mapping))
+        
+        # Test empty mapping is falsy
+        from python.runfiles.runfiles import _RepositoryMapping  # buildifier: disable=bzl-visibility
+        empty_mapping = _RepositoryMapping({}, {})
+        self.assertFalse(empty_mapping)
+        self.assertFalse(bool(empty_mapping))
+
+    # TODO: Add manifest-based test for compact repo mapping
```

Comment:
I'm not sure what this is referring to? The manifest and repo mapping files are different.